clippy_lints/src/utils/author.rs

```diff
@@ -190,17 +193,37 @@ impl PrintVisitor {
             println!("]);");
         }
     }
+
+    fn print_label(&mut self, label: Option<Label>) {
+        if let Some(label) = label {
+            let label_bind = self.next("label");
+
+            println!("    if let Some(ref {}) = {}", label_bind, self.current);
+
+            let label_name_bind = self.next("label_name");
+            let label_name = label.ident.name;
+
+            println!(
+                "    if {}.ident.name.as_str() == {:?};",
+                label_name_bind,
+                label_name.as_str()
+            );
+        } else {
+            println!("    if {}.is_none();", self.current);
+        }
+    }
 }
 
-struct PrintVisitor {
+struct PrintVisitor<'a, 'tcx> {
     /// Fields are the current index that needs to be appended to pattern
     /// binding names
     ids: FxHashMap<&'static str, usize>,
     /// the name that needs to be destructured
     current: String,
+    cx: Option<&'a LateContext<'tcx>>,
```

Comment:
Why is this optional?

Reply:
I was only using it in `visit_expr`, but I guess it doesn't really matter.